src/ashell/ide-comms.c

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2016-2017, Intel Corporation.
+
+/**
+ * @file
+ * @brief Device-side service for Web IDE message exchanges based on WebUSB.
+ *
+ */
+
+// C includes
+#include <ctype.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+// ZJS includes
+#include "ashell.h"
+#include "ide-comms.h"
+
+// Process a buffer (part of a message) in WebUSB driver context.
+void ide_receive(u8_t *buffer, size_t len)
+{
+    static char rx_buffer[P_SPOOL_SIZE + 1];
+    static size_t rx_cursor = 0;
+    extern void ide_parse(char *buf, size_t len);
+    char *buf = (char *) buffer;
+
+    if (len == 1) {  // character mode; buffer a command
+        ide_send_buffer(buf, 1);  // echo
+        if (rx_cursor >= P_SPOOL_SIZE || *buf == '\n' || *buf == '\r') {
+            //IDE_DBG("\r\nReceived %u chars.\r\n", rx_cursor);
```

Comment:
Please remove all commented IDE_DBG prints

Reply:
Of course I will, but I still need them for the raw webusb testing. This PR was made as a fallback (on WebUSB+UART), to give opportunity to early tests/comments etc. When the WiP flag is removed, all these will be cleaned up.